File: LibraryV0/Backend/library-management-system/src/main/java/com/library/lms/service/BookService.java
// path: Backend/src/main/java/com/library/lms/service/BookService.java
package com.library.lms.service;

import java.util.List;
import com.library.lms.model.Book;

public interface BookService {
    Book createBook(Book book);
    Book updateBook(Integer bookId, Book book);
    Book getBookById(Integer bookId);
    List<Book> getAllBooks();
    void deleteBook(Integer bookId);
}

File: LibraryV0/Frontend/src/services/bookService.js
// path: Frontend/src/services/BookService.js
import { api } from "./api.js";

// Helper to get JWT headers
const getAuthHeader = () => {
  const token = localStorage.getItem("token");
  return { Authorization: `Bearer ${token}` };
};

// --------------------
// Public Endpoints
// --------------------
export const getAllPublicBooks = async () => {
  const response = await api.get("/books");
  return response.data;
};

export const getBookById = async (id) => {
  const response = await api.get(`/books/${id}`);
  return response.data;
};

// --------------------
// Librarian-only Endpoints
// --------------------
export const getAllBooks = async () => {
  const response = await api.get("/books", { headers: getAuthHeader() });
  return response.data;
};

export const createBook = async (book) => {
  const response = await api.post("/books", book, { headers: getAuthHeader() });
  return response.data;
};

export const updateBook = async (id, book) => {
  const response = await api.put(`/books/${id}`, book, {
    headers: getAuthHeader(),
  });
  return response.data;
};

export const deleteBook = async (id) => {
  const response = await api.delete(`/books/${id}`, {
    headers: getAuthHeader(),
  });
  return response.data;
};
